staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -179,6 +186,7 @@ type WatchServer struct {
 	// used to encode the nested object in the watch stream
 	EmbeddedEncoder runtime.Encoder
 
+	MemoryAllocator      runtime.MemoryAllocator
```

Comment:
I'm looking for ways to do this change more incrementally and this seems to be a good one - can you move it to a separate PR (together with the first commit)?

In general - the simpler the change is, the easier it is to get convinced that it's correct.